src/platform/MpiServer.c

```diff
@@ -0,0 +1,611 @@
+                }
+                else
+                {
+                    OsConfigLogError(GetPlatformLog(), "Failed to allocate memory for response");
+                    status = HTTP_INTERNAL_SERVER_ERROR;
+                    *responseSize = 0;
+                }
+
+                FREE_MEMORY(uuid);
+            }
+            else
+            {
+                OsConfigLogError(GetPlatformLog(), "Failed to open session for client '%s'", clientName);
+                status = HTTP_INTERNAL_SERVER_ERROR;
+            }
+        }
+        else
+        {
+            OsConfigLogError(GetPlatformLog(), "Failed to parse request");
+            status = HTTP_BAD_REQUEST;
+        }
+
+        json_value_free(rootValue);
+    }
+    else
+    {
+        OsConfigLogError(GetPlatformLog(), "Failed to parse request");
+        status = HTTP_BAD_REQUEST;
+    }
+
+    return status;
+}
+
+static HTTP_STATUS MpiCloseRequest(const char* request, char** response, int* responseSize)
+{
+    HTTP_STATUS status = HTTP_OK;
+    JSON_Value* rootValue = NULL;
+    JSON_Object* rootObject = NULL;
+    JSON_Value* clientSessionValue = NULL;
+    char* clientSession = NULL;
+
+    UNUSED(response);
+    UNUSED(responseSize);
+
+    if ((NULL != (rootValue = json_parse_string(request))) && (NULL != (rootObject = json_value_get_object(rootValue))))
+    {
+        clientSessionValue = json_object_get_value(rootObject, g_clientSession);
+
+        if (clientSessionValue && (JSONString == json_value_get_type(clientSessionValue)))
+        {
+            clientSession = (char*)json_value_get_string(clientSessionValue);
+
+            if (IsFullLoggingEnabled())
+            {
+                OsConfigLogInfo(GetPlatformLog(), "Received MpiClose request, session '%s'", clientSession);
+            }
+
+            MpiClose((MPI_HANDLE)clientSession);
+        }
+        else
+        {
+            OsConfigLogError(GetPlatformLog(), "Failed to parse '%s' from request", g_clientSession);
+            status = HTTP_BAD_REQUEST;
+        }
+
+        json_value_free(rootValue);
+    }
+    else
+    {
+        OsConfigLogError(GetPlatformLog(), "Failed to parse request");
+        status = HTTP_BAD_REQUEST;
+    }
+
+    return status;
+}
+
+static HTTP_STATUS MpiSetRequest(const char* request, char** response, int* responseSize)
+{
+    HTTP_STATUS status = HTTP_OK;
+    JSON_Value* rootValue = NULL;
+    JSON_Object* rootObject = NULL;
+    JSON_Value* clientSessionValue = NULL;
+    JSON_Value* componentValue = NULL;
+    JSON_Value* objectValue = NULL;
+    JSON_Value* payloadValue = NULL;
+    int mpiStatus = MPI_OK;
+    char* clientSession = NULL;
+    char* component = NULL;
+    char* object = NULL;
+    char* payload = NULL;
+    int estimatedSize = 0;
+    const char* responseFormat = "\"%d\"";
+
+    if ((NULL != (rootValue = json_parse_string(request))) && (NULL != (rootObject = json_value_get_object(rootValue))))
```

Comment:
>if ((NULL != (rootValue = json_parse_string(request))) && (NULL != (rootObject = json_value_get_object(rootValue)))) [](http://example.com/codeflow?start=4&length=116)

Same comment on splitting this line to have proper logging for errors. Will stop here.